airbyte-cdk/python/airbyte_cdk/sources/streams/utils/stream_helper.py

```diff
@@ -0,0 +1,36 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, Mapping, Optional
+
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams.core import Stream, StreamData
+
+
+def get_first_stream_slice(stream) -> Optional[Mapping[str, Any]]:
+    """
+    Gets the first stream_slice from a given stream's stream_slices.
+    :param stream: stream
+    :return: First stream slice from 'stream_slices' generator
+    """
+    # We wrap the return output of stream_slices() because some implementations return types that are iterable,
+    # but not iterators such as lists or tuples
+    slices = iter(
+        stream.stream_slices(
+            cursor_field=stream.cursor_field,
+            sync_mode=SyncMode.full_refresh,
+        )
+    )
+    return next(slices)
+
+
+def get_first_record_for_slice(stream: Stream, stream_slice: Optional[Mapping[str, Any]]) -> StreamData:
```

Comment:
is it worth mentioning this will raise an exception if there are no records?
alternatively, this could return an `Optional[StreamData]`

Reply:
I tried to name it so that it is more intuitive, but clearly not intuitive enough if we're asking the question! will add it to the function description

Reply:
Optional would imply that `None` is a valid record, which I don't think is what we want